commons/ava_services/gecko_service_initializer_core.go

```diff
@@ -114,9 +115,10 @@ func (g GeckoServiceInitializerCore) InitializeMountedFiles(osFiles map[string]*
 
 	certFilePointer := osFiles[stakingTlsCertFileId]
 	keyFilePointer := osFiles[stakingTlsKeyFileId]
-	if len(dependencies) == 0 {
-		certFilePointer.WriteString(STAKER_1_CERT)
-		keyFilePointer.WriteString(STAKER_1_PRIVATE_KEY)
+	defaultStakers := ava_default_testnet.DefaultTestNet.DefaultStakers
+	if len(dependencies) < 5 {
```

Comment:
Ah I don't think this section is quite right - as-is, this code would say, "If you declare a node with < 5 dependencies (no matter what it is) then use the defaultStakers as the certs. If you had a node chain of bootstrapper <- child1 <- child2 <- child3, then len(dependencies) == 1 in every case and each node would get the same cert of `defaultStakers[0]`

Reply:
That `len(dependencies) == 0` was our way of doing "you're a bootstrapper node" in the past (because 0 dependencies == bootstrapper historically).

However: the work I'm doing for dupe node ID stuff is to refactor this initializer core to take in a TlsCertKeyProvider, which allows you to modify how the service config will assign certs; this will refactor this chunk of code (super glad that your stuff came in first). If we want to merge this code as-is and I can fix this up in my PR with the TlsCertKeyProvider, can we add a big fat TODO this is a big hack?

Reply:
merge code as-is, i'll add a big fat (the fattest) TODO

Reply:
done